### PR TITLE
Added support for translation to Brazilian Portuguese.

### PR DESCRIPTION
**Pull request by NeverAgain11**
#569

### CODE REVIEW

**Comment by NeverAgain11** [comment]
@tisfeng 

新增支持翻译到巴西葡萄牙语时，用 pt-BR 时请求翻译出现错误.

Bind 翻译的错误是:

bing json parse failed
{
  "errorMessage" : "",
  "statusCode" : 400
}}

DeepL 的错误是:

Request failed: bad request (400)
Invalid Request

能帮忙看看是什么问题么？

**Comment by tisfeng** [comment]
ok，明天我看看。

**Comment by Jerry23011** [comment]
截屏2024-06-13 01 37 00
DeepL works on my end

**Comment by Jerry23011** [comment]
As for Bing, it just doesn't seem to support `pt-BR`?
截屏2024-06-13 02 00 12
截屏2024-06-13 02 00 47

**Comment by NeverAgain11** [comment]
@Jerry23011 Could you try to translate from Chinese text to pt-BR?

**Comment by Jerry23011** [comment]
> @Jerry23011 Could you try to translate from Chinese text to pt-BR?

When targeting pt-BR, DeepL says Invalid target_lang
```json
{
  "jsonrpc": "2.0",
  "id": 150256000,
  "error": {
    "code": -32600,
    "message": "Invalid Request",
    "data": {
      "what": "Invalid target_lang."
    }
  }
}
```

**Comment by tisfeng** [comment]
The official API of DeepL supports PT-BR language, but the web version of our API does not support it.

image

**Comment by tisfeng** [comment]
I just added DeepL support for regional variant languages, like pt-BR 1cc75611

image

**Comment by tisfeng** [comment]
Note that I just debugged the Bing web API and found that the current code for the Portuguese language is wrong, it's a bit strange, for Portuguese Portuguese it's `pt-PT`, for Brazilian Portuguese it's `pt` 😓

I have fixed it adde77e1 .

[Bing Brazilian Portuguese translate](https://www.bing.com/translator/?text=%E5%85%AB%E5%8D%81%E4%B8%83%E5%B9%B4%E5%89%8D%EF%BC%8C%E6%88%91%E4%BB%AC%E7%9A%84%E7%88%B6%E8%BE%88%E4%BB%AC%E5%9C%A8%E8%BF%99%E5%9D%97%E9%99%86%E5%9C%B0%E4%B8%8A%E5%BB%BA%E8%B5%B7%E4%BA%86%E4%B8%80%E4%B8%AA%E6%96%B0%E7%9A%84%E5%9B%BD%E5%AE%B6%EF%BC%8C%E5%AE%83%E5%AD%95%E8%82%B2%E4%BA%8E%E8%87%AA%E7%94%B1%EF%BC%8C%E5%B9%B6%E5%A5%89%E8%A1%8C%E4%BA%BA%E4%BA%BA%E7%94%9F%E8%80%8C%E5%B9%B3%E7%AD%89%E7%9A%84%E4%B8%BB%E5%BC%A0%E3%80%82&from=zh-Hans&to=pt-BR)



image

image

**Review by tisfeng — CHANGES_REQUESTED** [review]
Please check if other services support Brazilian Portuguese language, such as Baidu Translate, it is supported.

image

File: Easydict/objc/Service/Language/EZLanguageModel.m [hunk 1]
```diff
@@ -156,9 +157,18 @@ @implementation EZLanguageModel
         portuguese.localName = @"Português";
         portuguese.flagEmoji = @"🇵🇹";
         portuguese.localeIdentifier = @"pt_PT";
-        portuguese.voiceName = @"Luciana";
+        portuguese.voiceName = @"Joana";
```
Comment (tisfeng) [inline Easydict/objc/Service/Language/EZLanguageModel.m:160]: Here Portugal has the wrong voiceName, I changed it to `Joana`.

A lot of this code is sample code, I wrote it with the help of ChatGPT, so I can't guarantee that it's correct 😥
 
image

**Comment by Jerry23011** [comment]
The latest changes works on my end

**Comment by NeverAgain11** [comment]
I checked other services and added support for Baidu, Google and Niu.
The Google translation has the same effect when using pt, pt-PT and pt-BR.

**Comment by tisfeng** [comment]
> The Google translation has the same effect when using pt, pt-PT and pt-BR.

It seems Google Translate doesn't support pt-BR, and the web version doesn't have the Brazilian Portuguese language option either.

**Review by tisfeng — APPROVED** [review]

**Pull request merged**